operator/pkg/reconciliation/construct_podtemplatespec.go

```diff
@@ -0,0 +1,481 @@
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	ServerConfigContainerName = "server-config-init"
+	CassandraContainerName    = "cassandra"
+	PvcName                   = "server-data"
+	SystemLoggerContainerName = "server-system-logger"
+)
+
+// calculateNodeAffinity provides a way to pin all pods within a statefulset to the same zone
+func calculateNodeAffinity(zone string) *corev1.NodeAffinity {
+	if zone == "" {
+		return nil
+	}
+	return &corev1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+			NodeSelectorTerms: []corev1.NodeSelectorTerm{
+				{
+					MatchExpressions: []corev1.NodeSelectorRequirement{
+						{
+							Key:      "failure-domain.beta.kubernetes.io/zone",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{zone},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// calculatePodAntiAffinity provides a way to keep the db pods of a statefulset away from other db pods
+func calculatePodAntiAffinity(allowMultipleNodesPerWorker bool) *corev1.PodAntiAffinity {
+	if allowMultipleNodesPerWorker {
+		return nil
+	}
+	return &corev1.PodAntiAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+			{
+				LabelSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key:      api.ClusterLabel,
+							Operator: metav1.LabelSelectorOpExists,
+						},
+						{
+							Key:      api.DatacenterLabel,
+							Operator: metav1.LabelSelectorOpExists,
+						},
+						{
+							Key:      api.RackLabel,
+							Operator: metav1.LabelSelectorOpExists,
+						},
+					},
+				},
+				TopologyKey: "kubernetes.io/hostname",
+			},
+		},
+	}
+}
+
+func selectorFromFieldPath(fieldPath string) *corev1.EnvVarSource {
+	return &corev1.EnvVarSource{
+		FieldRef: &corev1.ObjectFieldSelector{
+			FieldPath: fieldPath,
+		},
+	}
+}
+
+func probe(port int, path string, initDelay int, period int) *corev1.Probe {
+	return &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Port: intstr.FromInt(port),
+				Path: path,
+			},
+		},
+		InitialDelaySeconds: int32(initDelay),
+		PeriodSeconds:       int32(period),
+	}
+}
+
+func getJvmExtraOpts(dc *api.CassandraDatacenter) string {
+	flags := ""
+
+	if dc.Spec.DseWorkloads.AnalyticsEnabled == true {
+		flags += "-Dspark-trackers=true "
+	}
+	if dc.Spec.DseWorkloads.GraphEnabled == true {
+		flags += "-Dgraph-enabled=true "
+	}
+	if dc.Spec.DseWorkloads.SearchEnabled == true {
+		flags += "-Dsearch-service=true"
+	}
+	return flags
+}
+
+func combineVolumeMountSlices(defaults []corev1.VolumeMount, overrides []corev1.VolumeMount) []corev1.VolumeMount {
+	out := append([]corev1.VolumeMount{}, overrides...)
+outerLoop:
+	// Only add the defaults that don't have an override
+	for _, volumeDefault := range defaults {
+		for _, volumeOverride := range overrides {
+			if volumeDefault.Name == volumeOverride.Name {
+				continue outerLoop
```

Comment:
let's use a `found := false` boolean instead of the named continue

Reply:
Ok.  I had that initially and it is more boilerplate-ish, but I can easily switch back.

Reply:
nvm